src/main/scala/uk/gov/nationalarchives/omega/api/services/LocalProducer.scala

```diff
@@ -214,14 +221,22 @@ class LocalProducerImpl(val jmsProducer: JmsProducer[IO], val outputQueue: Queue
     localMessage: LocalMessage,
     outgoingMessageType: OutgoingMessageType,
     jsonErrors: Json
-  ): IO[Unit] =
-    localMessage.jmsMessageId
-      .filter(_.trim.nonEmpty)
-      .map(requestMessageId => sendError(requestMessageId, outgoingMessageType, jsonErrors))
-      .getOrElse(IO.unit)
+  ): IO[Unit] = {
+    val result: Option[(String, String)] = for {
+      messageId    <- localMessage.jmsMessageId
+      replyAddress <- localMessage.omgReplyAddress
+    } yield (messageId, replyAddress)
+    result match {
+      case Some((messageId, replyAddress)) =>
+        sendError(messageId, QueueName(replyAddress), outgoingMessageType, jsonErrors)
+      case _ =>
+        logger.error(s"Unable to reply to message ${localMessage.persistentMessageId} due to insufficient information.")
+    }
```

Comment:
could this error message be specific like `...due to missing messageId or replyAddress`

Reply:
Updated.